### DISCUSSION TITLE
Migration of big codebase from akka to pekko

### DISCUSSION THREAD

**Discussion by darkjh** [apache/pekko#1440] [Q&A]
Hello, 

I'm planning to migrate a big java codebase from akka 2.6.21 to pekko 1.0.0. To give an idea, there's over ~1800 files containing akka symbol import. 
The obvious option is to rename every import from `akka.*` to `org.apache.pekko.*` and adapt any config overrides in the codebase. However there's many active contributors to this codebase and this approach will create a lot of conflicts during the testing phase. 

I'm wondering has someone tried "smarter" approach to avoid the import changes? I tried using maven shade plugin to relocate classes from `org.apache.pekko` to `akka` and adapts some config values, and this seems to work for small examples. Maybe we can also intercept class loading and proxy the akka classes to pekko ones? 

Has anyone tried these "smarter" ways and are they worth pursuing? Thanks!

**Comment by pjfanning** [comment]
Using a shading tool might work but it is complicated. Remember that the config property names have changed too. Some class names that had 'Akka' in the name were changed (to have 'Pekko' in the name instead). Another issue to consider is that not every lib that supports Akka has been converted to support Pekko.

I would still suggest that you consider just using the vanilla approach of changing your code. Many people have gone down this route and we have had few complaints. Text replacement, especially using IDEs, is a pretty painless way to approach the problem. Updating PRs that are in flight is usually pretty easy with modern tooling (to rebase them).

**Comment by raboof** [comment]
Is this a Java project or a Scala project? If it's Scala, I wonder if you could build and share a [scalafix](https://scalacenter.github.io/scalafix/) rule to do the first pass of the migration.

Shading seems hard to get right and leave you with considerable technical depth even if it works...
